GSA/digitalgov.gov
# Header refactoring

## Summary

Modularize header files in a way similar to the current configuration of `guide-header.html`, implement a `header-stack.html` component, which consists of the title/deck/kicker.

## Preview

[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/)

## Solution

The solution implements the `header-stack.html` partial, which handle two new parameters `append_community` and `hide_kicker`. The `append_community` parameter appends "Community" to the title, and the `hide_kicker` parameter conditionally hides the kicker text. The community page template was updated to utilize these parameters. These parameters are needed to assure the community template can maintain the unique aspects of its appearance while still adhering to the modular setup of the components.

`header-stack` renders the common elements of the headers, the title, kicker, and deck. Parameters for hiding specific aspects of the component are included to allow for a flexible setup.

The `resources-header` and `communities-header` are modeled after the `guides-header` setup which separates the header from the `single.html` file. Changes consist solely of refactoring and have no change in appearance.

## How To Test
| Description              | Original URL                                                                                   | Updated URL                                                                                                               |
|--------------------------|------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------|
| Communities              | [Original Communities](https://digital.gov/communities/communicators/)                         | [Updated Communities](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/communities/communicators/) |
| Event                    | [Original Event](https://digital.gov/event/2024/06/20/uswds-monthly-call-june-2024/)            | [Updated Event](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/event/2024/06/20/uswds-monthly-call-june-2024/) |
| Resources                | [Original Resources](https://digital.gov/topics/accessibility/)                                | [Updated Resources](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/topics/accessibility/) |
| News                     | [Original News](https://digital.gov/2024/08/01/supporting-innovation-through-the-american-rescue-plan/) | [Updated News](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/Header-Refactoring/2024/08/01/supporting-innovation-through-the-american-rescue-plan/) |
| Guides                   | [Original Guides](https://digital.gov/guides/web-analytics-playbook/)                          | [Updated Guides](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/refactor-usa/guides/web-analytics-playbook/) |


Before opening this PR, make sure you’ve done whichever of these applies to you:
- [ ] Branch is up-to-date and includes latest from `main`
